reflex/app.py

```diff
@@ -563,6 +568,71 @@ def compile(self):
         thread_pool.close()
         thread_pool.join()
 
+    @contextlib.asynccontextmanager
+    async def modify_state(self, token: str) -> AsyncIterator[State]:
```

Comment:
This function doesn't actually modify the state right? Should we call it `get_latest_state` or something?

Reply:
the intent is that you modify the state inside the context and then it saves it to the backing store when you exit the context. it's a combination of `get_state` and `set_state`, with exclusive locking baked in.